mobile/lib/screens/map/map_view_widgets.dart

```diff
@@ -158,47 +158,44 @@ class SearchTile extends StatelessWidget {
 
   Future<void> _showPlaceDetails(BuildContext context) async {
     loadingScreen(context);
-
-    SearchResult? place = await SearchApiClient().getPlaceDetails(
+    await SearchApiClient()
+        .getPlaceDetails(
       searchResult,
-    );
-
-    if (place != null) {
-      final nearestSite = await LocationService.getNearestSite(
-        place.latitude,
-        place.longitude,
-      );
-
-      Navigator.pop(context);
-
-      // TODO: Substitute with widget
-      if (nearestSite == null) {
+    )
+        .then((place) async {
+      if (place != null) {
+        await LocationService.getNearestSite(
+          place.latitude,
+          place.longitude,
+        ).then((nearestSite) async {
```

Comment:
The map search functionality should work

Reply:
thanks Noah!